src/antares/tsgen/ts_generator.py

```diff
@@ -66,54 +92,90 @@ def _check_array(condition: npt.NDArray[np.bool_], message: str) -> None:
-    if cluster.unit_count <= 0:
-        raise ValueError(f"Unit count must be strictly positive, got {cluster.unit_count}.")
     if cluster.nominal_power <= 0:
         raise ValueError(f"Nominal power must be strictly positive, got {cluster.nominal_power}.")
-    if cluster.fo_volatility < 0:
-        raise ValueError(f"Forced outage volatility must be positive, got {cluster.unit_count}.")
-    if cluster.po_volatility < 0:
-        raise ValueError(f"Planned outage volatility must be positive, got {cluster.unit_count}.")
-
-    _check_1_dim(cluster.fo_rate, "Forced outage failure rate")
-    _check_1_dim(cluster.fo_duration, "Forced outage duration")
-    _check_1_dim(cluster.po_rate, "Planned failure rate")
-    _check_1_dim(cluster.po_duration, "Planned outage duration")
-    _check_1_dim(cluster.npo_min, "Minimum count of planned outages")
-    _check_1_dim(cluster.npo_max, "Maximum count of planned outages")
+
+    _check_outage_gen_params(cluster.outage_gen_params)
+
     _check_1_dim(cluster.modulation, "Hourly modulation")
+
     if len(cluster.modulation) != 8760:
         raise ValueError("hourly modulation array must have 8760 values.")
 
-    _check_array(cluster.fo_rate < 0, "Forced failure rate is negative on following days")
-    _check_array(cluster.fo_rate > 1, "Forced failure rate is greater than 1 on following days")
-    _check_array(cluster.fo_duration <= 0, "Forced outage duration is null or negative on following days")
-    _check_array(cluster.po_rate < 0, "Planned failure rate is negative on following days")
-    _check_array(cluster.po_rate > 1, "Planned failure rate is greater than 1 on following days")
-    _check_array(cluster.po_duration <= 0, "Planned outage duration is null or negative on following days")
     _check_array(cluster.modulation < 0, "Hourly modulation is negative on following hours")
 
+    lengths = _check_lengths(cluster.outage_gen_params)
+    if len(lengths) != 1:
+        raise ValueError(f"Not all daily arrays have same size, got {lengths}")
+
+
+def _check_link_capacity(link_capacity: LinkCapacity) -> None:
+    if link_capacity.nominal_capacity <= 0:
+        raise ValueError(f"Nominal power must be strictly positive, got {link_capacity.nominal_capacity}.")
+
+    _check_outage_gen_params(link_capacity.outage_gen_params)
+
+    _check_1_dim(link_capacity.modulation_indirect, "Direct modulation")
+    _check_1_dim(link_capacity.modulation_indirect, "Indirect hourly modulation")
+
+    if len(link_capacity.modulation_direct) != 8760 or len(link_capacity.modulation_indirect) != 8760:
+        raise ValueError("hourly modulation array must have 8760 values.")
+
+    _check_array(link_capacity.modulation_direct < 0, "Hourly direct modulation is negative on following hours")
+    _check_array(link_capacity.modulation_indirect < 0, "Hourly indirect modulation is negative on following hours")
+
+    lengths = _check_lengths(link_capacity.outage_gen_params)
+
+    if len(lengths) != 1:
```

Comment:
You could also add this inside the `_check_lenghts` method you've created it would be nice :)